dittofs-operator/internal/controller/config/config.go

```diff
@@ -0,0 +1,251 @@
+package config
+
+import (
+	"fmt"
+
+	dittoiov1alpha1 "github.com/marmos91/dittofs/dittofs-operator/api/v1alpha1"
+
+	"gopkg.in/yaml.v3"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func GenerateDittoFSConfig(dittoServer *dittoiov1alpha1.DittoServer) (string, error) {
+	metadataStores := make(map[string]MetadataStore)
+	contentStores := make(map[string]ContentStore)
+
+	for _, backend := range dittoServer.Spec.Config.Backends {
+		switch backend.Type {
+		case "badger":
+			metadataStores[backend.Name] = MetadataStore{
+				Type: "badger",
+				Badger: &BadgerConfig{
+					DBPath: getConfigValue(backend.Config, "path", "/data/metadata"),
+				},
+			}
+		case "local":
+			contentStores[backend.Name] = ContentStore{
+				Type: "filesystem",
+				Filesystem: &FilesystemConfig{
+					Path: getConfigValue(backend.Config, "path", "/data/content"),
+				},
+			}
+		case "s3":
+			contentStores[backend.Name] = ContentStore{
+				Type: "s3",
+				S3: &S3Config{
+					Bucket:          getConfigValue(backend.Config, "bucket", ""),
+					Region:          getConfigValue(backend.Config, "region", "us-east-1"),
+					Endpoint:        getConfigValue(backend.Config, "endpoint", ""),
+					AccessKeyID:     getConfigValue(backend.Config, "access_key_id", ""),
+					SecretAccessKey: getConfigValue(backend.Config, "secret_access_key", ""),
+					ForcePathStyle:  getConfigValue(backend.Config, "force_path_style", "") == "true",
```

Comment:
For now it's ok like this. We should manage these by secrets though using `secretKeyRef`. Let's not forget about this. Would you like to create an issue to track this task?

Reply:
Absolutely yes, I will open it once it's merged so that I will be able to reference this line of code with a permalink.